src/hooks/validation/models.ts

```diff
@@ -0,0 +1,113 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+interface Specification {
+  isNotValid(): boolean;
+  read(): string;
+  getSpecificationName(): string;
+}
+
+export class SpecificationFile implements Specification {
+  private readonly name: string;
+
+  constructor(name: string) {
+    const localDirectory = process.cwd();
+    this.name = path.join(localDirectory, name);
+  }
+
+  isNotValid(): boolean {
+    return !fs.existsSync(this.name) || !fs.lstatSync(this.name).isFile();
+  }
+
+  read(): string {
+    return fs.readFileSync(this.name, 'utf8');
+  }
+
+  getSpecificationName(): string {
+    return this.name;
+  }
+}
+
+export type WatchFlag = boolean;
+
+export class ValidationResponse {
+  private readonly _success: Boolean;
+  private readonly _errors: Array<String>;
+
+  constructor(success: Boolean, errorsInfo: String[] = []) {
+    this._success = success;
+    this._errors = errorsInfo;
+  }
+
+  get success(): Boolean {
+    return this._success;
+  }
+
+  get errors(): Array<String> {
+    return this._errors;
+  }
+
+  public static createSuccess(): ValidationResponse {
+    return new ValidationResponse(true);
+  }
+
+  public static createError(err: any): ValidationResponse {
+    const errorsInfo: Array<String> = [];
+
+    if (err.title) {
+      errorsInfo.push(err.title);
+    }
+    if (err.detail) {
+      errorsInfo.push(err.detail);
+    }
+    if (err.validationErrors) {
+      err.validationErrors.forEach((e: any) =>
+        errorsInfo.push(`${e.title} ${e.location.startLine}:${e.location.startColumn}`)
```

Comment:
unfortunately, it can still happen that location is not provided, so I can get an error like `Cannot read property 'startLine' of undefined` for some errors.

you need protection to check if `location` is part of validation error or not 🤷🏼

Reply:
oops! I will change it!! 😋

Reply:
done